### PR TITLE
Update proof of stake section (wiki migration)

### PR DESCRIPTION
**Pull request by jmcook1186**
## Description

This PR is part of the migration of materials from eth.wiki to ethereum.org. There is substantial material on eth.wiki concerning proof of stake that was not available at ethereum.org and should not be lost when the wiki is deprecated. The PoS page currently on ethereum.org is also rather sparse and makes frequent reference to sharding designs that have since been deprecated. For these reasons I have reworked the PoS landing page with up-to-date information and also added two new subpages: one on weak subjectivity and another explaining the Gasper protocol. This satisfies the migration of PoS materials from eth.wiki. Note, however, that the material there was written when Casper was the cutting edge of PoS research, meaning it lags significantly behind the mechanism that will be implemented at the merge. Therefore, I have rewritten and updated where necessary.

In summary the following changes are implemented:

-  Substantial rewriting of PoS landing page, includign removal of OOD information
- Add new "Gasper" page that cointains high-level explanation of Casper and LMD-GHOST as they will be implemented at the merge
- Add "weak subjectivity" page. This is part of the wiki migration but content rewritten to better fit ethereum.org

Note also that, contrary to the suggestions in the wiki migration issue I decided NOT to migrate the PoS FAQs onto ethereum.org. This is because they are already replicated verbatim on vitalik.ca. I have simply linked out to that page instead.

## Related Issue
#5690

### CODE REVIEW

File: src/content/developers/docs/consensus-mechanisms/pos/index.md
Comment (minimalsm) [inline src/content/developers/docs/consensus-mechanisms/pos/index.md:21]: Could we use proof-of-work / proof-of-stake over the acronym [as per the style guide](https://ethereum.org/en/contributing/style-guide/#proof-of)

File: src/content/developers/docs/consensus-mechanisms/pos/gasper/index.md
Comment (minimalsm) [inline src/content/developers/docs/consensus-mechanisms/pos/gasper/index.md:13]: Could you please add header ids to any new headings 🙏
```suggestion
## The role of Gasper {#gasper}
```

File: src/content/developers/docs/consensus-mechanisms/pos/weak-subjectivity/index.md
Comment (minimalsm) [inline src/content/developers/docs/consensus-mechanisms/pos/weak-subjectivity/index.md:15]: Thoughts on switching these around and removing the weak subjectivity heading? We don't explicitly state it anywhere (we should), but this seems like the convention we lean towards most often: ([for example](https://ethereum.org/en/developers/docs/development-networks/))

```suggestion
Subjectivity in blockchains refers to reliance upon social information to agree on the current state. There may be multiple valid forks that are chosen from according to information gathered from other peers on the network. The converse is objectivity which refers to chains where there is only one possible valid chain that all nodes will necessarily agree upon by applying their coded rules. There is also a third state, known as weak subjectivity. This refers to a chain that can progress objectively after some initial seed of information is retrieved socially.

## Prerequisites {#prerequisites}

To understand this page it is necessary to first understand the fundamentals of [proof-of-stake](/developers/docs/consensus-mechanisms/pos/).
```

**Review by minimalsm — COMMENTED** [review]
Hey @jmcook1186, this looks great ❤️❤️. Left some high-level comments but will try to do a full review by tomorrow.

File: src/content/developers/docs/consensus-mechanisms/pos/weak-subjectivity/index.md
Comment (jmcook1186) [reply src/content/developers/docs/consensus-mechanisms/pos/weak-subjectivity/index.md]: ok done in next commit

File: src/content/developers/docs/consensus-mechanisms/pos/gasper/index.md
Comment (jmcook1186) [reply src/content/developers/docs/consensus-mechanisms/pos/gasper/index.md]: done

File: src/content/developers/docs/consensus-mechanisms/pos/index.md
Comment (jmcook1186) [reply src/content/developers/docs/consensus-mechanisms/pos/index.md:21]: done

File: src/content/developers/docs/consensus-mechanisms/pos/weak-subjectivity/index.md
Comment (minimalsm) [inline src/content/developers/docs/consensus-mechanisms/pos/weak-subjectivity/index.md:21]:
```suggestion
Weak subjectivity is implemented in proof-of-stake Ethereum by using "weak subjectivity checkpoints". These are state roots that all nodes on the network agree belong in the canonical chain. They serve the same "universal truth" purpose to genesis blocks, except that they do not sit at the genesis position in the blockchain. The fork choice algorithm trusts that the blockchain state defined in that checkpoint is correct and that it independently and objectively verifies the chain from that point onwards. The checkpoints act as "revert limits" because blocks located before weak-subjectivity checkpoints cannot be changed. This undermines long-range attacks simply by defining long-range forks to be invalid as part of the mechanism design. Ensuring that the weak subjectivity checkpoints are separated by a smaller distance than the validator withdrawal period ensures that a validator that forks the chain is slashed at least some threshold amount before they can withdraw their stake and that new entrants cannot be tricked onto incorrect forks by validators whose stake has been withdrawn.
```

File: src/content/developers/docs/consensus-mechanisms/pos/weak-subjectivity/index.md
Comment (minimalsm) [inline src/content/developers/docs/consensus-mechanisms/pos/weak-subjectivity/index.md:25]:
```suggestion
Finalized blocks and weak subjectivity checkpoints are treated differently by Ethereum nodes. If a node becomes aware of two competing finalized blocks, then it is torn between the two - it has no way to identify automatically which is the canonical fork. This is symptomatic of a consensus failure. In contrast, a node simply rejects any block that conflicts with its weak subjectivity checkpoint. From the node's perspective, the weak subjectivity checkpoint represents an absolute truth that cannot be undermined by new knowledge from its peers.
```

File: src/content/developers/docs/consensus-mechanisms/pos/weak-subjectivity/index.md
Comment (minimalsm) [inline src/content/developers/docs/consensus-mechanisms/pos/weak-subjectivity/index.md:33]:
```suggestion
The subjective aspect of Ethereum's proof-of-stake is the requirement for a recent state (weak subjectivity checkpoint) from a trusted source to sync from. The risk of getting a bad weak subjectivity checkpoint is very low because they can be checked against several independent public sources such as block explorers or multiple nodes. However, there is always some degree of trust required to run any software application, for example, trusting that the software developers have produced honest software.

A weak subjectivity checkpoint may even come as part of the client software. Arguably an attacker can corrupt the checkpoint in the software and can just as easily corrupt the software itself. There is no real crypto-economic route around this problem, but the impact of untrustworthy developers is minimized in Ethereum by having multiple independent client teams, each building equivalent software in different languages, all with a vested interest in maintaining an honest chain. Block explorers may also provide weak subjectivity checkpoints or a way to cross-reference checkpoints obtained from elsewhere against an additional source.

Finally, checkpoints can be requested from other nodes; perhaps another Ethereum user that runs a full node can provide a checkpoint that validators can then verify against data from a block explorer. Overall, trusting the provider of a weak subjectivity checkpoint can be considered as problematic as trusting the client developers. The overall trust required is low. It is important to note that these considerations only become important in the very unlikely event that a majority of validators conspire to produce an alternate fork of the blockchain. Under any other circumstances, there is only one Ethereum chain to choose from.
```

File: src/content/developers/docs/consensus-mechanisms/pos/gasper/index.md
Comment (minimalsm) [inline src/content/developers/docs/consensus-mechanisms/pos/gasper/index.md:3]:
```suggestion
description: An explanation of the Gasper proof-of-stake mechanism.
```

**Review by minimalsm — COMMENTED** [review]
This is great @jmcook1186 (And I actually learned a bunch, thanks 😀)! I've committed some formatting changes but nothing major.

Left two comments for you to consider but not blocking this PR; we can definitely merge this and work on those later.

I also think we want to consider where we want these to live as they seem a little bit beyond 'Foundational' (definitely not something to worry about just now, we can do this at a later date).

File: src/content/developers/docs/consensus-mechanisms/pos/index.md
Comment (minimalsm) [reply src/content/developers/docs/consensus-mechanisms/pos/index.md]: Hmmm I had to fix a bunch of these and realised that this maybe isn't actually clear:

We stick to spelling out full terms (proof-of-stake) over acronyms (PoS) to avoid the potential cognitive overhead it causes.

See [this section of the Google Technical Writing Guide](https://developers.google.com/tech-writing/one/words?hl=en#use_the_acronym_or_the_full_term) for the reasoning we used to inform this choice.

File: src/content/developers/docs/consensus-mechanisms/pos/gasper/index.md
Comment (wackerow) [inline src/content/developers/docs/consensus-mechanisms/pos/gasper/index.md:11]:
```suggestion
**Note** that the original definition of Casper-FFG was updated slightly for inclusion in Gasper. On this page we consider the updated version.
```

File: src/content/developers/docs/consensus-mechanisms/pos/gasper/index.md
Comment (wackerow) [inline src/content/developers/docs/consensus-mechanisms/pos/gasper/index.md:19]: Small note, but feel like we should open this section with a sentence directly answering "What is finality?" before explaining more about Casper, etc.

File: src/content/developers/docs/consensus-mechanisms/pos/gasper/index.md
Comment (wackerow) [inline src/content/developers/docs/consensus-mechanisms/pos/gasper/index.md:35]: >  The rewards ether added to their stake. 

Attempt at correcting:
```suggestion
Validators get rewarded for honestly proposing and validating blocks. Ether is rewarded and added to their stake. On the other hand, validators that are absent and fail to act when called upon miss out on these rewards and sometimes lose a small portion of their existing stake. However, the penalties for being offline are small and, in most cases, amount to opportunity costs of missing rewards. However, some validator actions are very difficult to do accidentally and signify some malicious intent, such as proposing multiple blocks for the same slot, attesting to multiple blocks for the same slot, or contradicting previous checkpoint votes. These are "slashable" behaviors that are penalized more harshly—slashing results in some portion of the validator's stake being destroyed and the validator being removed from the network of validators. This process takes 36 days. On Day 1, there is an initial penalty of up to 0.5 ETH. Then the slashed validator's ether slowly drains away across the exit period, but on Day 18, they receive a "correlation penalty", which is larger when more validators are slashed around the same time. The maximum penalty is the entire stake. These rewards and penalties are designed to incentivize honest validators and disincentivize attacks on the network.
```

File: src/content/developers/docs/consensus-mechanisms/pos/index.md
Comment (wackerow) [inline src/content/developers/docs/consensus-mechanisms/pos/index.md:9]: > demonstrably more secure than PoW

*If implemented correctly. Would we want to say this has been demonstrated already given that we still haven't undergone the merge? Or should we note it as theoretical?

File: src/content/developers/docs/consensus-mechanisms/pos/index.md
Comment (wackerow) [inline src/content/developers/docs/consensus-mechanisms/pos/index.md:32]: Perhaps switch the order of mentioning committee's and block proposers. Ie:
```suggestion
Whereas under proof-of-work, the timing of blocks is determined by the mining difficulty, in proof-of-stake, the tempo is fixed. Time in proof-of-stake Ethereum is divided into slots (12 seconds) and epochs (32 slots). One validator is randomly selected to be a block proposer in every slot. This validator is responsible for creating a new block and sending it out to other nodes on the network. Also in every slot, a committee of validators is randomly chosen, whose votes are used to determine the validity of the block being proposed.
```

File: src/content/developers/docs/consensus-mechanisms/pos/index.md
Comment (wackerow) [inline src/content/developers/docs/consensus-mechanisms/pos/index.md:44]: Should we link out to the Gasper page on some of these terms? ie. LMD-GHOST, or "inactivity leak" above?

File: src/content/developers/docs/consensus-mechanisms/pos/index.md
Comment (wackerow) [inline src/content/developers/docs/consensus-mechanisms/pos/index.md:52]: Similar question as above... Are we considering the Beacon Chain to have demonstrated this in its current form?

File: src/content/developers/docs/consensus-mechanisms/pos/index.md
Comment (wackerow) [inline src/content/developers/docs/consensus-mechanisms/pos/index.md:59]: Would argue more hash power = more returns just like more staked ETH = more returns... The thing to note here is PoS doesn't really enable economies of scale, as ASICs are expensive, and you can get significant discounts when buying bulk... and opportunity the little guy cannot take advantage of. PoS has reasonably low fixed costs (1 computer) but after that, the rate of return on your ETH is basically the same.

File: src/content/developers/docs/consensus-mechanisms/pos/index.md
Comment (wackerow) [reply src/content/developers/docs/consensus-mechanisms/pos/index.md:63]: Agreed 👍🏻

**Review by wackerow — COMMENTED** [review]
@jmcook1186 This is great! Love to see the expansion of these resources within the repo. 
Left a few small comments/suggestions/questions, but overall think this is great and looking forward to pulling it in!

File: src/content/developers/docs/consensus-mechanisms/pos/gasper/index.md
Comment (jmcook1186) [reply src/content/developers/docs/consensus-mechanisms/pos/gasper/index.md:10]: done

File: src/content/developers/docs/consensus-mechanisms/pos/gasper/index.md
Comment (jmcook1186) [reply src/content/developers/docs/consensus-mechanisms/pos/gasper/index.md]: yup - done

File: src/content/developers/docs/consensus-mechanisms/pos/index.md
Comment (jmcook1186) [reply src/content/developers/docs/consensus-mechanisms/pos/index.md:9]: ok, changed "is demonstrably" to "is thought to be"

File: src/content/developers/docs/consensus-mechanisms/pos/index.md
Comment (jmcook1186) [reply src/content/developers/docs/consensus-mechanisms/pos/index.md:59]: fair point. updated accordingly.

File: src/content/developers/docs/consensus-mechanisms/pos/index.md
Comment (jmcook1186) [reply src/content/developers/docs/consensus-mechanisms/pos/index.md:63]: done

**Comment by minimalsm** [comment]
Just waiting on another review before merging.

File: src/content/developers/docs/consensus-mechanisms/pos/gasper/index.md
Comment (samajammin) [inline src/content/developers/docs/consensus-mechanisms/pos/gasper/index.md:6]: ✂️ - see comment below.

File: src/content/developers/docs/consensus-mechanisms/pos/weak-subjectivity/index.md
Comment (samajammin) [inline src/content/developers/docs/consensus-mechanisms/pos/weak-subjectivity/index.md:6]: ✂️

File: src/content/developers/docs/consensus-mechanisms/pos/index.md
Comment (samajammin) [inline src/content/developers/docs/consensus-mechanisms/pos/index.md:9]: Perhaps out of the scope of this PR (since this is already how the page is currently) but I wonder if it'd be best to first have a quick intro on what a consensus mechanism is & what PoS is before getting into how it relates specifically to the Ethereum roadmap?

Comment (samajammin) [reply src/content/developers/docs/consensus-mechanisms/pos/index.md:9]: Just thinking that people often land on this page when searching "proof of stake". Many people may be looking for a general explanation before diving into the specifics on Ethereum.

**Review by wackerow — APPROVED** [review]
💪🏻

**Pull request merged**
